README.md

```diff
@@ -12,62 +12,370 @@ Install from NPM:
+  await hoverElem(elem);
+  // do assertions
+});
+
+it('should click on element', async() => {
+  const elem = await fixture(...);
+  await clickElem(elem);
+  // do assertions
+});
+```
+
+Alternatively, to hover over or click at viewport coordinates with the mouse, use `hoverAt(x, y)` or `clickAt(x, y)`:
+
+```javascript
+import { clickAt, hoverAt } from '@brightspace-ui/testing';
+
+it('should hover at coordinate', async() => {
+  await hoverAt(100, 200);
+  // do assertions
+});
+
+it('should click at coordinate', async() => {
+  await clickAt(100, 200);
+  // do assertions
+});
+```
+
+### Using the Keyboard
+
+To place focus on an element using the keyboard, use `focusElem(elem)`. Doing so will trigger its `:focus-visible` CSS pseudo-class.
+
+```javascript
+import { fixture, focusElem } from '@brightspace-ui/testing';
+
+it('should focus on element', async() => {
+  const elem = await fixture(html`<button>elem</button>`);
+  await focusElem(elem);
+  // do assertions
+});
+```
+
+To send particular keystrokes to the browser window or a specific element, use `sendKeys(action, keys)` or `sendKeysElem(elem, action, keys)`.
+
+The `action` parameter must be one of:
+
+* `type`: types a sequence of characters and **is not affected** by modifier keys such as holding down `Shift`
+* `press`: presses a single key, which results in a `keydown` followed by a `keyup` and **is affected** by modifier keys such as `Shift`
+* `down`: holds down a single key
+* `up`: releases a single key
+
+For a list of all available key values, refer to [key values for keyboard events](https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_key_values).
+
+```javascript
+import { fixture, sendKeysElem } from '@brightspace-ui/testing';
+
+it('should type and press Enter', async() => {
+  const elem = await fixture(html`<input type="text">`);
+  await sendKeysElem(elem, 'type', 'Hello');
+  await sendKeysElem(elem, 'press', 'Enter');
+});
+```
+
+As demonstrated above, subsequent calls can be made to create key combinations.
+
+### Waiting for Things
+
+There are various scenarios where a test may need to wait before it can proceed.
+
+#### Waiting for an Event
+
+After interacting with components, to wait for a particular event to be dispatched use `oneEvent(elem, eventName)`.
+
+```javascript
+import { clickElem, fixture, oneEvent } from '@brightspace-ui/testing';
+
+it('should wait for an event', async() => {
+  const elem = await fixture(html`<my-elem></my-elem>`);
+  clickElem(elem);
+  await oneEvent(elem, 'some-event');
+});
+```
 
-# unit tests only
-npm run test:headless
+Note that the call to `clickElem` is not `await`-ed, since by the time it resolves the event will likely have already been dispatched.
+
+An even safer approach would be to wrap that call in a `setTimeout`:
+
+```javascript
+setTimeout(() => clickElem(elem));
+await oneEvent(elem, 'some-event');
+```
+
+#### Waiting for a Lit Element to Update
```

Comment:
This could easily be left out, but it could be useful to call it out in case folks use `aTimeout` or `nextFrame` instead of realizing they can do this? The [open-wc stuff](https://open-wc.org/docs/testing/helpers/#elementupdated) has a special `await elementUpdated(elem)` that essentially does this and falls back to `nextFrame` if it's not a Lit element.